vllm/model_executor/parameter.py

```diff
@@ -0,0 +1,191 @@
+from typing import Callable, List, Optional, Tuple, Union
+
+import torch
+from torch.nn import Parameter
+
+from vllm.distributed import get_tensor_model_parallel_rank
+from vllm.logger import init_logger
+
+__all__ = [
+    "vLLMParameter", "PackedvLLMParameter", "PerTensorScaleParameter",
+    "ModelWeightParameter", "ChannelQuantScaleParameter",
+    "GroupQuantScaleParameter"
+]
+
+logger = init_logger(__name__)
+
+
+class vLLMParameter(Parameter):
+
+    def __new__(cls, data: torch.Tensor, **kwargs):
+
+        return super().__new__(cls, data=data, requires_grad=False)
+
+    def __init__(self,
+                 data: torch.Tensor,
+                 weight_loader: Callable,
+                 ignore_warnings: Optional[bool] = True):
+
+        self._ignore_warnings = True
+        self._weight_loader = weight_loader
+
+    @property
+    def weight_loader(self):
+        return self._weight_loader
+
+
+class _ColumnvLLMParameter(vLLMParameter):
```

Comment:
I find multiple levels of inheritance to be hard to follow.

There is definitely some re-used code here though. 

Perhaps we could have some floating functions that are shared across the subclasses?